packages/supplementary-contracts/contracts/TokenUnlocking.sol

```diff
@@ -3,233 +3,126 @@ pragma solidity 0.8.24;
+    event TokenWithdrawn(address to, uint256 amount);
 
-    error INVALID_GRANTEE();
     error INVALID_PARAM();
-    error WRONG_GRANTEE_RECIPIENT();
+    error NOT_WITHDRAWABLE();
+    error PERMISSION_DENIED();
+    error TOO_LATE();
+
+    modifier onlyRecipient() {
+        if (msg.sender != recipient) revert PERMISSION_DENIED();
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
 
     /// @notice Initializes the contract.
     /// @param _owner The contract owner address.
     /// @param _taikoToken The Taiko token address.
-    /// @param _sharedVault The shared vault address.
-    /// @param _grantRecipient Who will be the grantee for this contract.
+    /// @param _recipient Who will be the grantee for this contract.
+    /// @param _tgeTimestamp The token generation event timestamp.
     function init(
         address _owner,
         address _taikoToken,
-        address _sharedVault,
-        address _grantRecipient,
+        address _recipient,
         uint64 _tgeTimestamp
     )
         external
         initializer
     {
-        if (
-            _taikoToken == address(0) || _sharedVault == address(0) || _grantRecipient == address(0)
-                || _tgeTimestamp == 0
-        ) {
+        if (_taikoToken == address(0) || _recipient == address(0) || _tgeTimestamp == 0) {
             revert INVALID_PARAM();
         }
 
-        // OZ 4.9.6. version does not allow param setting with __Ownable_init(), so we transfer the
-        // ownership afterwards.
-        __Ownable_init();
         _transferOwnership(_owner);
 
+        recipient = _recipient;
         taikoToken = _taikoToken;
-        sharedVault = _sharedVault;
-
-        // Initializing here, that the contract belongs to this grant recipient, and TGE starts or
-        // started at _tgeTimestamp.
-        grantRecipient = _grantRecipient;
         tgeTimestamp = _tgeTimestamp;
-
-        emit GrantInitialized(
-            _grantRecipient, _tgeTimestamp, getCliffEndTimestamp(), getUnlockPeriod()
-        );
     }
 
-    /// @notice Triggers a deposits through the vault to this contract.
-    /// This transaction should happen on a regular basis, e.g.: quarterly.
-    /// @param _recipient The grant recipient address.
-    /// @param _currentDeposit The current deposit.
-    function vestToken(
-        address _recipient,
-        uint128 _currentDeposit
-    )
-        external
-        onlyOwner
-        nonReentrant
-    {
-        if (_recipient != grantRecipient) revert INVALID_GRANTEE();
+    /// @notice Vest certain tokens to this contract.
+    /// @param _amount The newly vested amount
+    function vest(uint128 _amount) external onlyOwner nonReentrant {
+        if (_amount == 0) revert INVALID_PARAM();
+        if (block.timestamp >= tgeTimestamp + FOUR_YEARS) revert TOO_LATE();
```

Comment:
I got the point, that after TGE + 4 years, ppl can just pay and get their tokens instantly (without this contract), but seems like than no point having a check here, since it is not "dangerous" if the transfer would go thru this contract anyway. But ok.

Reply:
will remove it.